### PR TITLE
Clarified meaning of 'generic' and 'generic type'

### LINKED ISSUE #3715: Review usage of "generic type" in System.Runtime.InteropServices.Marshal.xml file
# Review usage of "generic type" in System.Runtime.InteropServices.Marshal.xml file

The documentation for the [Marshal class](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal?view=netframework-4.7.1) and its members uses the phrase "generic type" in 14 instances. In some cases, that usage is incorrect, and "generic type" should be changed to "generic type definition." 

See #3671 for additional context.

//cc @cleak

### PR DESCRIPTION
**Pull request by rpetrusha**
# Clarified meaning of 'generic' and 'generic type'

## Summary
There were two problems here:
- "Generic" was used before the introduction of generics to describe a base class.
- "Generic type" was used loosely without clarifying whether the term referred to a generic type definition or an instance of a generic type.

This PR addresses those issues.

Fixes #3715 

## Suggested Reviewers

@mairaw @BillWagner @cleak

### CODE REVIEW

**Review by jzeferino — COMMENTED** [review]
LGTM.

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 1]
```diff
@@ -7578,7 +7578,7 @@
   
  -or-  
   
- <paramref name="structure" /> is a generic type.</exception>
+ <paramref name="structure" /> is an instance of a generic type (the .NET Framework 4.5 and earlier versions only). </exception>
```
Comment (mairaw) [inline xml/System.Runtime.InteropServices/Marshal.xml]: nit: should this have a preposition like in the .NET Framework 4.5?

**Review by mairaw — APPROVED** [review]
LGTM. Left a small question for you.

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 2]
```diff
@@ -3492,7 +3492,7 @@
...
+ The `pUnk` parameter represents an `IUnknown` interface pointer; however, because all COM interfaces derive directly or indirectly from `IUnknown`, you can pass any COM interface to this method. The object returned by <xref:System.Runtime.InteropServices.Marshal.GetObjectForIUnknown%2A> is a [Runtime Callable Wrapper](~/docs/framework/interop/runtime-callable-wrapper.md), which the common language runtime manages as it does any other managed object. The type of this wrapper is often a base `System.__ComObject` type, which is a hidden type used when the wrapper type is ambiguous. You can still make late-bound calls to such a base type as long as the COM object implements the [IDispatch](http://msdn.microsoft.com/en-us/ebbff4bc-36b2-4861-9efa-ffa45e013eb5) interface. Likewise, you can cast the returned object to an appropriate COM interface.
```
Comment (mairaw) [inline xml/System.Runtime.InteropServices/Marshal.xml]: unrelated to your change but this topic is the only one using this link to the IDispatch interface that makes you have to click again to find the right topic. I can fix it later.

Comment (rpetrusha) [reply xml/System.Runtime.InteropServices/Marshal.xml]: I've fixed it, @mairaw.

**Pull request merged**